engine/src/deploy_macosx.cpp

```diff
@@ -1088,6 +1135,55 @@ static void relocate_function_starts_command(linkedit_data_command *x, int32_t p
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static bool MCDeployToMacOSXFetchMinOSVersion(const MCDeployParameters& p_params, mach_header& p_header, uint32_t& r_version)
+{
+    // First work out what DeployArchitecture to look for.
```

Comment:
It needs to be a cascaded if since some cpu-types depend on two fields to determine - however, I did notice that I wasn't setting 'Unknown' in the case of no match - so have fixed that.